File: binding/src/libqcalendar.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qcalendar.html
pub const qcalendar = struct {

    /// New constructs a new QCalendar object.
    pub fn New(other: ?*anyopaque) QtC.QCalendar {
        return qtc.QCalendar_new(@ptrCast(other));
    }


    /// New2 constructs a new QCalendar object.
    pub fn New2(other: ?*anyopaque) QtC.QCalendar {
        return qtc.QCalendar_new2(@ptrCast(other));
    }


    /// New3 constructs a new QCalendar object.
    pub fn New3() QtC.QCalendar {
        return qtc.QCalendar_new3();
    }


    /// New4 constructs a new QCalendar object.
    pub fn New4(system: i32) QtC.QCalendar {
        return qtc.QCalendar_new4(@intCast(system));
    }


    /// New5 constructs a new QCalendar object.
    pub fn New5(id: QtC.QCalendar__SystemId) QtC.QCalendar {
        return qtc.QCalendar_new5(@ptrCast(id));
    }


    /// New6 constructs a new QCalendar object.
    pub fn New6(param1: ?*anyopaque) QtC.QCalendar {
        return qtc.QCalendar_new6(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCalendar_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCalendar_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QCalendar_IsValid(@ptrCast(self));
    }

    pub fn DaysInMonth(self: ?*anyopaque, month: i32) i32 {
        return qtc.QCalendar_DaysInMonth(@ptrCast(self), month);
    }

    pub fn DaysInYear(self: ?*anyopaque, year: i32) i32 {
        return qtc.QCalendar_DaysInYear(@ptrCast(self), year);
    }

    pub fn MonthsInYear(self: ?*anyopaque, year: i32) i32 {
        return qtc.QCalendar_MonthsInYear(@ptrCast(self), year);
    }

    pub fn IsDateValid(self: ?*anyopaque, year: i32, month: i32, day: i32) bool {
        return qtc.QCalendar_IsDateValid(@ptrCast(self), year, month, day);
    }

    pub fn IsLeapYear(self: ?*anyopaque, year: i32) bool {
        return qtc.QCalendar_IsLeapYear(@ptrCast(self), year);
    }

    pub fn IsGregorian(self: ?*anyopaque, ) bool {
        return qtc.QCalendar_IsGregorian(@ptrCast(self));
    }

    pub fn IsLunar(self: ?*anyopaque, ) bool {
        return qtc.QCalendar_IsLunar(@ptrCast(self));
    }

    pub fn IsLuniSolar(self: ?*anyopaque, ) bool {
        return qtc.QCalendar_IsLuniSolar(@ptrCast(self));
    }

    pub fn IsSolar(self: ?*anyopaque, ) bool {
        return qtc.QCalendar_IsSolar(@ptrCast(self));
    }

    pub fn IsProleptic(self: ?*anyopaque, ) bool {
        return qtc.QCalendar_IsProleptic(@ptrCast(self));
    }

    pub fn HasYearZero(self: ?*anyopaque, ) bool {
        return qtc.QCalendar_HasYearZero(@ptrCast(self));
    }

    pub fn MaximumDaysInMonth(self: ?*anyopaque, ) i32 {
        return qtc.QCalendar_MaximumDaysInMonth(@ptrCast(self));
    }

    pub fn MinimumDaysInMonth(self: ?*anyopaque, ) i32 {
        return qtc.QCalendar_MinimumDaysInMonth(@ptrCast(self));
    }

    pub fn MaximumMonthsInYear(self: ?*anyopaque, ) i32 {
        return qtc.QCalendar_MaximumMonthsInYear(@ptrCast(self));
    }

    pub fn Name(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_Name(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.Name: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn DateFromParts(self: ?*anyopaque, year: i32, month: i32, day: i32) QtC.QDate {
        return qtc.QCalendar_DateFromParts(@ptrCast(self), year, month, day);
    }

    pub fn DateFromParts2(self: ?*anyopaque, parts: ?*anyopaque) QtC.QDate {
        return qtc.QCalendar_DateFromParts2(@ptrCast(self), @ptrCast(parts));
    }

    pub fn MatchCenturyToWeekday(self: ?*anyopaque, parts: ?*anyopaque, dow: i32) QtC.QDate {
        return qtc.QCalendar_MatchCenturyToWeekday(@ptrCast(self), @ptrCast(parts), dow);
    }

    pub fn PartsFromDate(self: ?*anyopaque, date: QtC.QDate) QtC.QCalendar__YearMonthDay {
        return qtc.QCalendar_PartsFromDate(@ptrCast(self), @ptrCast(date));
    }

    pub fn DayOfWeek(self: ?*anyopaque, date: QtC.QDate) i32 {
        return qtc.QCalendar_DayOfWeek(@ptrCast(self), @ptrCast(date));
    }

    pub fn MonthName(self: ?*anyopaque, locale: ?*anyopaque, month: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_MonthName(@ptrCast(self), @ptrCast(locale), month);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.MonthName: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn StandaloneMonthName(self: ?*anyopaque, locale: ?*anyopaque, month: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_StandaloneMonthName(@ptrCast(self), @ptrCast(locale), month);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.StandaloneMonthName: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn WeekDayName(self: ?*anyopaque, locale: ?*anyopaque, day: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_WeekDayName(@ptrCast(self), @ptrCast(locale), day);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.WeekDayName: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn StandaloneWeekDayName(self: ?*anyopaque, locale: ?*anyopaque, day: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_StandaloneWeekDayName(@ptrCast(self), @ptrCast(locale), day);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.StandaloneWeekDayName: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn AvailableCalendars() []const u8 {
        return qtc.QCalendar_AvailableCalendars();
    }

    pub fn DaysInMonth2(self: ?*anyopaque, month: i32, year: i32) i32 {
        return qtc.QCalendar_DaysInMonth2(@ptrCast(self), month, year);
    }

    pub fn MonthName3(self: ?*anyopaque, locale: ?*anyopaque, month: i32, year: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_MonthName3(@ptrCast(self), @ptrCast(locale), month, year);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.MonthName3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn MonthName4(self: ?*anyopaque, locale: ?*anyopaque, month: i32, year: i32, format: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_MonthName4(@ptrCast(self), @ptrCast(locale), month, year, @intCast(format));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.MonthName4: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn StandaloneMonthName3(self: ?*anyopaque, locale: ?*anyopaque, month: i32, year: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_StandaloneMonthName3(@ptrCast(self), @ptrCast(locale), month, year);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.StandaloneMonthName3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn StandaloneMonthName4(self: ?*anyopaque, locale: ?*anyopaque, month: i32, year: i32, format: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_StandaloneMonthName4(@ptrCast(self), @ptrCast(locale), month, year, @intCast(format));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.StandaloneMonthName4: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn WeekDayName3(self: ?*anyopaque, locale: ?*anyopaque, day: i32, format: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_WeekDayName3(@ptrCast(self), @ptrCast(locale), day, @intCast(format));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.WeekDayName3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn StandaloneWeekDayName3(self: ?*anyopaque, locale: ?*anyopaque, day: i32, format: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCalendar_StandaloneWeekDayName3(@ptrCast(self), @ptrCast(locale), day, @intCast(format));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcalendar.StandaloneWeekDayName3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCalendar_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qcalendar-yearmonthday.html
pub const qcalendar__yearmonthday = struct {

    /// New constructs a new QCalendar::YearMonthDay object.
    pub fn New(other: ?*anyopaque) QtC.QCalendar__YearMonthDay {
        return qtc.QCalendar__YearMonthDay_new(@ptrCast(other));
    }


    /// New2 constructs a new QCalendar::YearMonthDay object.
    pub fn New2(other: ?*anyopaque) QtC.QCalendar__YearMonthDay {
        return qtc.QCalendar__YearMonthDay_new2(@ptrCast(other));
    }


    /// New3 constructs a new QCalendar::YearMonthDay object.
    pub fn New3() QtC.QCalendar__YearMonthDay {
        return qtc.QCalendar__YearMonthDay_new3();
    }


    /// New4 constructs a new QCalendar::YearMonthDay object.
    pub fn New4(y: i32) QtC.QCalendar__YearMonthDay {
        return qtc.QCalendar__YearMonthDay_new4(y);
    }


    /// New5 constructs a new QCalendar::YearMonthDay object.
    pub fn New5(y: i32, m: i32) QtC.QCalendar__YearMonthDay {
        return qtc.QCalendar__YearMonthDay_new5(y, m);
    }


    /// New6 constructs a new QCalendar::YearMonthDay object.
    pub fn New6(y: i32, m: i32, d: i32) QtC.QCalendar__YearMonthDay {
        return qtc.QCalendar__YearMonthDay_new6(y, m, d);
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCalendar__YearMonthDay_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCalendar__YearMonthDay_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QCalendar__YearMonthDay_IsValid(@ptrCast(self));
    }

    pub fn Year(self: ?*anyopaque, ) i32 {
        return qtc.QCalendar__YearMonthDay_Year(@ptrCast(self));
    }

    pub fn SetYear(self: ?*anyopaque, year: i32) void {
        qtc.QCalendar__YearMonthDay_SetYear(@ptrCast(self), year);
    }

    pub fn Month(self: ?*anyopaque, ) i32 {
        return qtc.QCalendar__YearMonthDay_Month(@ptrCast(self));
    }

    pub fn SetMonth(self: ?*anyopaque, month: i32) void {
        qtc.QCalendar__YearMonthDay_SetMonth(@ptrCast(self), month);
    }

    pub fn Day(self: ?*anyopaque, ) i32 {
        return qtc.QCalendar__YearMonthDay_Day(@ptrCast(self));
    }

    pub fn SetDay(self: ?*anyopaque, day: i32) void {
        qtc.QCalendar__YearMonthDay_SetDay(@ptrCast(self), day);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCalendar__YearMonthDay_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qcalendar-systemid.html
pub const qcalendar__systemid = struct {

    /// New constructs a new QCalendar::SystemId object.
    pub fn New(other: ?*anyopaque) QtC.QCalendar__SystemId {
        return qtc.QCalendar__SystemId_new(@ptrCast(other));
    }


    /// New2 constructs a new QCalendar::SystemId object.
    pub fn New2(other: ?*anyopaque) QtC.QCalendar__SystemId {
        return qtc.QCalendar__SystemId_new2(@ptrCast(other));
    }


    /// New3 constructs a new QCalendar::SystemId object.
    pub fn New3() QtC.QCalendar__SystemId {
        return qtc.QCalendar__SystemId_new3();
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCalendar__SystemId_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QCalendar__SystemId_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Index(self: ?*anyopaque, ) u64 {
        return qtc.QCalendar__SystemId_Index(@ptrCast(self));
    }

    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QCalendar__SystemId_IsValid(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCalendar__SystemId_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/calendar.html#types
pub const enums = struct {
    pub const System = enum {
        pub const Gregorian: i32 = 0;
        pub const Julian: i32 = 8;
        pub const Milankovic: i32 = 9;
        pub const Jalali: i32 = 10;
        pub const IslamicCivil: i32 = 11;
        pub const Last: i32 = 11;
        pub const User: i32 = -1;
    };

};

File: binding/src/libqcalendar.cpp
#include <QCalendar>
#define WORKAROUND_INNER_CLASS_DEFINITION_QCalendar__SystemId
#define WORKAROUND_INNER_CLASS_DEFINITION_QCalendar__YearMonthDay
#include <QDate>
#include <QLocale>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qcalendar.h>
#include "libqcalendar.h"
#include "libqcalendar.hxx"

QCalendar* QCalendar_new(const QCalendar* other) {
	 return new QCalendar(*other);
}

QCalendar* QCalendar_new2(QCalendar* other) {
	 return new QCalendar(*other);
}

QCalendar* QCalendar_new3() {
	 return new QCalendar();
}

QCalendar* QCalendar_new4(int system) {
	 return new QCalendar(static_cast<QCalendar::System>(system));
}

QCalendar* QCalendar_new5(QCalendar__SystemId* id) {
	 return new QCalendar(*id);
}

QCalendar* QCalendar_new6(const QCalendar* param1) {
	 return new QCalendar(*param1);
}

void QCalendar_CopyAssign(QCalendar* self, QCalendar* other) {
    *self = *other;
}

void QCalendar_MoveAssign(QCalendar* self, QCalendar* other) {
    *self = std::move(*other);
}

bool QCalendar_IsValid(const QCalendar* self) {
	return self->isValid();
}

int QCalendar_DaysInMonth(const QCalendar* self, int month) {
	return self->daysInMonth(month);
}

int QCalendar_DaysInYear(const QCalendar* self, int year) {
	return self->daysInYear(year);
}

int QCalendar_MonthsInYear(const QCalendar* self, int year) {
	return self->monthsInYear(year);
}

bool QCalendar_IsDateValid(const QCalendar* self, int year, int month, int day) {
	return self->isDateValid(year, month, day);
}

bool QCalendar_IsLeapYear(const QCalendar* self, int year) {
	return self->isLeapYear(year);
}

bool QCalendar_IsGregorian(const QCalendar* self) {
	return self->isGregorian();
}

bool QCalendar_IsLunar(const QCalendar* self) {
	return self->isLunar();
}

bool QCalendar_IsLuniSolar(const QCalendar* self) {
	return self->isLuniSolar();
}

bool QCalendar_IsSolar(const QCalendar* self) {
	return self->isSolar();
}

bool QCalendar_IsProleptic(const QCalendar* self) {
	return self->isProleptic();
}

bool QCalendar_HasYearZero(const QCalendar* self) {
	return self->hasYearZero();
}

int QCalendar_MaximumDaysInMonth(const QCalendar* self) {
	return self->maximumDaysInMonth();
}

int QCalendar_MinimumDaysInMonth(const QCalendar* self) {
	return self->minimumDaysInMonth();
}

int QCalendar_MaximumMonthsInYear(const QCalendar* self) {
	return self->maximumMonthsInYear();
}

libqt_string QCalendar_Name(const QCalendar* self) {
	QString _ret = self->name();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QDate* QCalendar_DateFromParts(const QCalendar* self, int year, int month, int day) {
	return new QDate(self->dateFromParts(year, month, day));
}

QDate* QCalendar_DateFromParts2(const QCalendar* self, const QCalendar__YearMonthDay* parts) {
	return new QDate(self->dateFromParts(*parts));
}

QDate* QCalendar_MatchCenturyToWeekday(const QCalendar* self, const QCalendar__YearMonthDay* parts, int dow) {
	return new QDate(self->matchCenturyToWeekday(*parts, dow));
}

QCalendar__YearMonthDay* QCalendar_PartsFromDate(const QCalendar* self, QDate* date) {
	return new QCalendar::YearMonthDay(self->partsFromDate(*date));
}

int QCalendar_DayOfWeek(const QCalendar* self, QDate* date) {
	return self->dayOfWeek(*date);
}

libqt_string QCalendar_MonthName(const QCalendar* self, const QLocale* locale, int month) {
	QString _ret = self->monthName(*locale, month);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCalendar_StandaloneMonthName(const QCalendar* self, const QLocale* locale, int month) {
	QString _ret = self->standaloneMonthName(*locale, month);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCalendar_WeekDayName(const QCalendar* self, const QLocale* locale, int day) {
	QString _ret = self->weekDayName(*locale, day);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCalendar_StandaloneWeekDayName(const QCalendar* self, const QLocale* locale, int day) {
	QString _ret = self->standaloneWeekDayName(*locale, day);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_list QCalendar_AvailableCalendars() {
	auto _ret = QCalendar::availableCalendars();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		QByteArray _b = _ret[_i].toUtf8();
		libqt_string* _str = new libqt_string();
		_str->len = _b.length();
		_str->data = static_cast<const char*>(malloc(_str->len + 1));
		memcpy((void*)_str->data, _b.data(), _str->len);
		((char*)_str->data)[_str->len] = '\0';
		_data[_i] = _str;
	}
	return _arr;
}

int QCalendar_DaysInMonth2(const QCalendar* self, int month, int year) {
	return self->daysInMonth(month, year);
}

libqt_string QCalendar_MonthName3(const QCalendar* self, const QLocale* locale, int month, int year) {
	QString _ret = self->monthName(*locale, month, year);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCalendar_MonthName4(const QCalendar* self, const QLocale* locale, int month, int year, int format) {
	QString _ret = self->monthName(*locale, month, year, static_cast<QLocale::FormatType>(format));
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCalendar_StandaloneMonthName3(const QCalendar* self, const QLocale* locale, int month, int year) {
	QString _ret = self->standaloneMonthName(*locale, month, year);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCalendar_StandaloneMonthName4(const QCalendar* self, const QLocale* locale, int month, int year, int format) {
	QString _ret = self->standaloneMonthName(*locale, month, year, static_cast<QLocale::FormatType>(format));
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCalendar_WeekDayName3(const QCalendar* self, const QLocale* locale, int day, int format) {
	QString _ret = self->weekDayName(*locale, day, static_cast<QLocale::FormatType>(format));
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCalendar_StandaloneWeekDayName3(const QCalendar* self, const QLocale* locale, int day, int format) {
	QString _ret = self->standaloneWeekDayName(*locale, day, static_cast<QLocale::FormatType>(format));
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QCalendar_Delete(QCalendar* self) {
    delete self;
}

QCalendar__YearMonthDay* QCalendar__YearMonthDay_new(const QCalendar__YearMonthDay* other) {
	 return new QCalendar::YearMonthDay(*other);
}

QCalendar__YearMonthDay* QCalendar__YearMonthDay_new2(QCalendar__YearMonthDay* other) {
	 return new QCalendar::YearMonthDay(*other);
}

QCalendar__YearMonthDay* QCalendar__YearMonthDay_new3() {
	 return new QCalendar::YearMonthDay();
}

QCalendar__YearMonthDay* QCalendar__YearMonthDay_new4(int y) {
	 return new QCalendar::YearMonthDay(y);
}

QCalendar__YearMonthDay* QCalendar__YearMonthDay_new5(int y, int m) {
	 return new QCalendar::YearMonthDay(y, m);
}

QCalendar__YearMonthDay* QCalendar__YearMonthDay_new6(int y, int m, int d) {
	 return new QCalendar::YearMonthDay(y, m, d);
}

void QCalendar__YearMonthDay_CopyAssign(QCalendar__YearMonthDay* self, QCalendar__YearMonthDay* other) {
    *self = *other;
}

void QCalendar__YearMonthDay_MoveAssign(QCalendar__YearMonthDay* self, QCalendar__YearMonthDay* other) {
    *self = std::move(*other);
}

bool QCalendar__YearMonthDay_IsValid(const QCalendar__YearMonthDay* self) {
	return self->isValid();
}

int QCalendar__YearMonthDay_Year(const QCalendar__YearMonthDay* self) {
	return self->year;
}

void QCalendar__YearMonthDay_SetYear(QCalendar__YearMonthDay* self, int year) {
	self->year;
}

int QCalendar__YearMonthDay_Month(const QCalendar__YearMonthDay* self) {
	return self->month;
}

void QCalendar__YearMonthDay_SetMonth(QCalendar__YearMonthDay* self, int month) {
	self->month;
}

int QCalendar__YearMonthDay_Day(const QCalendar__YearMonthDay* self) {
	return self->day;
}

void QCalendar__YearMonthDay_SetDay(QCalendar__YearMonthDay* self, int day) {
	self->day;
}

void QCalendar__YearMonthDay_Delete(QCalendar__YearMonthDay* self) {
    delete self;
}

QCalendar__SystemId* QCalendar__SystemId_new(const QCalendar__SystemId* other) {
	 return new QCalendar::SystemId(*other);
}

QCalendar__SystemId* QCalendar__SystemId_new2(QCalendar__SystemId* other) {
	 return new QCalendar::SystemId(*other);
}

QCalendar__SystemId* QCalendar__SystemId_new3() {
	 return new QCalendar::SystemId();
}

void QCalendar__SystemId_CopyAssign(QCalendar__SystemId* self, QCalendar__SystemId* other) {
    *self = *other;
}

void QCalendar__SystemId_MoveAssign(QCalendar__SystemId* self, QCalendar__SystemId* other) {
    *self = std::move(*other);
}

unsigned long long QCalendar__SystemId_Index(const QCalendar__SystemId* self) {
	return self->index();
}

bool QCalendar__SystemId_IsValid(const QCalendar__SystemId* self) {
	return self->isValid();
}

void QCalendar__SystemId_Delete(QCalendar__SystemId* self) {
    delete self;
}

